### DISCUSSION TITLE
Question: recording and live at the same time

### DISCUSSION THREAD

**Discussion by stranger2904** [CESNET/UltraGrid#232] [Q&A]
Hello

Is it possible to setup a recording on a receiver instance ?
or record and transmit at the same time from the source ?
I've tried to use a combination of hd-rum-transcoder and ffmpeg but with no luck...

**Comment by alatteri** [comment]
You are trying to record the output stream or the input signal?

**Comment by stranger2904** [comment]
Output stream. My goal is to record locally high bitrate/high quality stream while send to receive side low-bitrate/low-res signal. hd-rum-transcoder allows me to split/downconvert signal but now I need to figure out how to make a recording.

**Comment by alatteri** [comment]
So what you are really trying to do is this:

```
         /-------low res stream -----------> to internet
INPUT -- 
         \------- high quality record file
```

**Comment by stranger2904** [comment]
yes, you are correct, this is what I'm trying to achieve

**Comment by MartinPulec** [comment]
Hi, both should be possible

> Is it possible to setup a recording on a receiver instance ?

`uv ... -d dump[:dir]`

(here there may not be another display, however)

> or record and transmit at the same time from the source ?

`uv ... --export[=dir] <recv>`

In both cases you'll get individual frames in the output directory. But it may not be a problem to post-process them, eg. for H.264 frames, simple concatenation gives playable/convertible stream, see also [here](https://github.com/CESNET/UltraGrid/wiki/Recording-and-Playback).
